server/routes/upload.js

```diff
@@ -0,0 +1,50 @@
+const express = require('express');
+const { NotExtended } = require('http-errors');
+const multer = require('multer');
+const cloudinary = require('cloudinary').v2;
+const streamifier = require('streamifier'); 
+const auth = require('../middleware/verifyAuth');
+const { GeneralError } = require('../utils/errors');
+const router = express.Router();
+const fileUpload = multer();
+
+//Get cloud_name, api_key and api_secret from .env
+cloudinary.config({ 
+    cloud_name: process.env.CLOUDINARY_CLOUD_NAME, 
+    api_key: process.env.CLOUDINARY_API_KEY,
+    api_secret: process.env.CLOUDINARY_API_SECRET
+});
+
+//stream file upload
+router.post("/", fileUpload.single("image"), auth, async function(req, res, next) {
+
+  try {
+    let streamUpload = (req) => {
```

Comment:
Get in the habit of using const for anything that shouldn't/isn't going to be overwritten.  There's a few cases in this file.